### DISCUSSION TITLE
Possible to have artillery call/use external API calls/tests from another framework

### DISCUSSION THREAD

**Discussion by humbienri** [artilleryio/artillery#965] [Artillery]
Hello all,

I've not been able to find any documentation on this other than a reference to calling custom JS. Perhaps what I am interested in trying to do is not possible?

I want to have artillery call and use some external (say supertest or axios) REST API tests and leverage those as part of a scenario in artillery.

I believe I read something about your function API? But I couldn't find any documentation on it. So in essence I am looking to inject or externalize the part your scenarios that are the actual POST or GET calls.

I hope that makes sense. I'm not a javascript developer.

**Comment by hassy** [comment]
Hi @humbienri👋 yes that should be possible with a bit of custom JS, e.g. with a `function` step in an HTTP scenario: https://artillery.io/docs/guides/guides/http-reference.html#Function-steps-in-a-flow

The API would let you call external tests, wait for their completion and then continue running the rest of the scenario. Note that metrics from HTTP calls made in the external test suite would _not_ be recorded by Artillery, but custom metrics may also be set by custom JS code via: https://artillery.io/docs/guides/guides/extending.html#Tracking-custom-metrics

**Comment by humbienri** [comment]
Thank you very much @hassy!

Interesting, I thought I already read through that documentation and it wasn't apparently obvious to me how it could be done. But, I'm new to this so I'll go look again. Thank you for the reference.

And yes, good reminder about the lack of metrics that would be available to Artillery for reporting.

Time to read some more. :)

**Reply by hassy** [reply]
If you can post a minimal example of an external test @humbienri (redacted / edited as needed), I can help with the glue code. It would make for a great addition to [artillery-examples](https://github.com/artilleryio/artillery-examples)

**Comment by humbienri** [comment]
Thanks man.

Let me come up with something, I'll need some time. Right now it's all reading since I'm not a JS person but in the past I did something similar as a proof of concept in the Python world. I'm trying to teach myself how to do it in JS so it'll be slow progress but I've been chomping at the bit, regarding Artillery, for a while now.

I just need to find a publicly available API that I could load hit for a brief minute.
